tests/server-integration/Common/BootServer.hs

```diff
@@ -0,0 +1,108 @@
+-- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+module Common.BootServer
+  ( unit_run_with_cmd_option_port
+  , unit_run_with_env_var_port
+  , unit_run_with_no_port_specified
+  , unit_run_with_cmd_opt_and_bad_env_var_port
+  , unit_run_with_bad_cmd_opt_and_env_var_port
+  , unit_run_with_bad_env_var_port
+  , unit_run_with_bad_cmd_option_port
+  , unit_run_with_bad_str_cmd_option_port
+  ) where
+
+import Control.Concurrent.Async (async, cancel, poll)
+import Control.Exception (SomeException(SomeException))
+import System.Environment (setEnv, unsetEnv, withArgs)
+import System.Time.Extra (sleep)
+import Test.Tasty.HUnit (assertFailure, (@=?))
+import Web.Main (RunServerException(..), runServer)
+
+testPort :: String
+testPort = "8079"
+
+maybeIO :: (a -> IO ()) -> Maybe a -> IO ()
+maybeIO action maybe =
```

Comment:
You can use `whenJust` from `Control.Monad.Extra`, instead of defining your own `maybeIO`.

Reply:
done